lib/api/model/model.dart

```diff
@@ -383,7 +383,7 @@ class DmRecipientListConverter extends JsonConverter<List<DmRecipient>, List<dyn
   }
 
   @override
-  List toJson(List<DmRecipient> object) => object;
+  List toJson(List<DmRecipient> object) => object.map((r) => r.toJson()).toList();
```

Comment:
Yeah, when I tried to convert a message into JSON and then back again, it failed. Apparently it was because inside the JSON, it had `"recipients": [ a Flutter DM Recipient Object ]` instead of what you'd expect, like `[ { "name": "some Name" ... }]`

Reply:
Ah, I see.

I think if you'd taken it all the way to a serialized JSON string, it'd work. (Because what `jsonEncode` does by default with something that's not a built-in type is to call its `toJson` method.) But it makes sense to want `toJson` and `fromJson` to be inverses on their own, without the extra step of going to actual serialized JSON and back.

Reply:
Aah yeah, fair point - that was exactly my assumption. I can leave it as it was, or keep this PR - obviously it'll be moot in the final version, so it's up to which you think makes more sense.

Reply:
Thinking a bit more, our many generated `toJson` methods don't do this, because they follow the default behavior here:
https://pub.dev/documentation/json_annotation/latest/json_annotation/JsonSerializable/explicitToJson.html

See for example `InitialSnapshot.toJson`, as that class is one that happens to have a bunch of fields containing non-primitive types.

So probably best to leave this out here, matching the generated methods. If we ever decide to flip that switch in the generated methods (perhaps because we find ourselves attempting a bunch of round trips), we can sweep through non-generated `toJson` implementations then to change their behavior accordingly.